src/sequence.rs

```diff
@@ -0,0 +1,155 @@
+use std::fs;
+
+use git2::Signature;
+use self::sfml::audio::Music;
+
+use Error;
+use utils;
+
+/// Creates a new user for the project in the current directory.
+/// Assumes the current directory contains a Protonfile.json file.
+///
+/// Impure.
+pub fn new_sequence<P: AsRef<Path>>(name: &str, music_file_path: P) -> Result<(), Error> {
+    
+    // Make sure the name is valid (needed since it will be used in a file path)
+    try!(verify_name(name));
+
+    // Make sure the music file is a valid format
+    try!(verify_file_type(&music_file_path));
+
+    // Make the name of the sequence's directory
+    let mut sequence_dir = String::from("seq_");
+    sequence_dir.push_str(&name);
+    let sequence_dir = sequence_dir;
+
+    // Try to create the sequence's directory
+    // This also throws an error if the directory already exists and is not empty
+    try!(utils::create_empty_directory(Path::new(&sequence_dir))
+        .map_err(|_| Error::DuplicateSequence(name.to_string()) ));
+
+    // Get name of music file from path
+    let music_file_name = try!(utils::file_name_from_path(&music_file_path));
+
+    // Try to copy music file into sequence directory
+    let dest_path = try!(copy_music_file(&music_file_path, &sequence_dir));
+
+    // Get duration of music file
+    let music_path_str = &music_file_path.as_ref().to_str().expect("Path is invalid");
+    let music_duration_sec = try!(get_music_duration_sec(&music_path_str));
+
+    // Add sequence to project
+    let project = try!(utils::read_protonfile(None::<P>));
+    let new_project = match project.add_sequence(
+        name,
+        &sequence_dir,
+        &music_file_name,
+        music_duration_sec
+    ) {
+        Ok(proj) => proj,
+        Err(e) => {
+            // Remove copied music file (clean up)
+            try!(fs::remove_file(&dest_path).map_err(Error::Io));
+            panic!(e.to_string())
+        },
+    };
+
+    // Save project
+    try!(utils::write_protonfile(&new_project, None::<P>));
+
+    // Commit changes
+    let signature = Signature::now("Proton Lights", "proton@teslaworks.net").unwrap();
+    let msg = format!("Adding new sequence '{}'", name);
+    let repo_path: Option<P> = None;
+
+    utils::commit_all(repo_path, &signature, &msg)
+        .map(|_| ())
+}
+
+/// Check that the music file is a valid format
+/// Full list of supported formats can be found at
+/// http://www.rust-sfml.org/doc/rsfml/audio/struct.Music.html
+fn verify_file_type<P: AsRef<Path>>(music_file_path: P) -> Result<(), Error> {
+    match music_file_path.as_ref().extension() {
+        Some(extension) => {
+            match extension.to_str() {
+                Some("ogg")  |
+                Some("wav")  |
+                Some("flac") |
+                Some("aiff") |
+                Some("raw") => Ok(()),
+                None => Err(unsupported_file_type_error("invalid extension")),
+                _ => Err(unsupported_file_type_error(extension.to_str().unwrap())),
```

Comment:
`Some(ex) => unsupported_file_type_error(ex)`. Saves on doing the conversion and unwrapping again.